crates/pessimistic-proof/src/bridge_exit.rs

```diff
@@ -31,6 +31,16 @@ pub enum LeafType {
     Message = 1,
 }
 
+impl LeafType {
+    pub fn from_u8(value: u8) -> Option<Self> {
```

Comment:
Why not implement `TryFrom` ?

Reply:
> Why not implement `TryFrom` ?

No big reason. It just requires coming up with an `Error` associated type which I found to be too much ceremony for such a little helper function.

Reply:
> Why not implement `TryFrom` ?

Changed it to `TryFrom`.